File: language-support/js/daml-ledger/src/validation/get_package_response.ts
// Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
// SPDX-License-Identifier: Apache-2.0

import * as ledger from "..";
import { native, object, Validation } from "./base";
import { RequiredValidation } from "./base/typelevel";
import { HashFunction } from "./hash_function";

function required(): RequiredValidation<ledger.GetPackageResponse> {
    return {
        archivePayload: native('string'),
        hash: native('string'),
        hashFunction: HashFunction
    };
}

export const GetPackageResponse: Validation = object<ledger.GetPackageResponse>('GetPackageResponse', required, () => ({}));

File: sdk/canton/community/ledger/ledger-json-api/src/main/scala/com/digitalasset/canton/http/admin/GetPackageResponse.scala
// Copyright (c) 2025 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
// SPDX-License-Identifier: Apache-2.0

package com.digitalasset.canton.http.admin

import com.daml.ledger.api.v2.package_service
import com.google.protobuf

sealed abstract class HashFunction extends Product with Serializable
case object SHA256 extends HashFunction
final case class Unrecognized(value: Int) extends HashFunction

object HashFunction {
  def fromLedgerApi(a: package_service.HashFunction): HashFunction = a match {
    case package_service.HashFunction.HASH_FUNCTION_SHA256 => SHA256
    case package_service.HashFunction.Unrecognized(x) => Unrecognized(x)
  }
}

final case class GetPackageResponse(
    hashFunction: HashFunction,
    hash: String,
    archivePayload: protobuf.ByteString,
)

object GetPackageResponse {
  def fromLedgerApi(a: package_service.GetPackageResponse): GetPackageResponse =
    GetPackageResponse(
      hashFunction = HashFunction.fromLedgerApi(a.hashFunction),
      archivePayload = a.archivePayload,
      hash = a.hash,
    )
}
